plugins/modules/aci_dhcp_relay.py

```diff
@@ -230,29 +272,40 @@ def main():
     tenant = module.params.get("tenant")
     child_classes = ["dhcpRsProv"]
 
-    aci = ACIModule(module)
-    aci.construct_url(
-        root_class=dict(
+    if tenant is None:
+        root_class = dict(
```

Comment:
why recreating the whole structure and not just changing the root class part to be class infra / rn infra / ... ?

Reply:
See answer in aci_dhcp_relay_provider module.